pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4208 @@
+                                   MaxObjectCount=None):
+        # pylint: disable=invalid-name
+        """
+        Implements WBEM server responder for
+        :meth:`~pywbem.WBEMConnection.OpenReferenceInstancePaths`
+        with data from the instance CIM repository.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            Name of the namespace in which the CIM instance will be created
+              (case independent).
+
+            Leading and trailing slash characters will be stripped. The lexical
+            case will be preserved.
+
+          InstanceName (:class:`~pywbem.CIMInstanceName`):
+            The instance path of the source instance.
+            If this object does specifies a namespace, it must equal the
+            value of the namespace parameter.
+            Its `host` attribute will be ignored.
+
+          See :meth:`~pywbem.WBEMConnection.ReferenceNames` for description of
+          ResultClass, Role and _open_response for description of other
+          parameters.
+
+        Returns:
+
+            See :meth:`~pywbem..WBEMConnection.ReferenceNames` for description
+            of the returns
+
+        Raises:
+
+            CIMError: CIM_ERR_INVALID_NAMESPACE
+            CIMError: CIM_ERR_INVALID_PARAMETER
+        """
+
+        self._test_pull_operations_disabled()
+        self._validate_open_params(FilterQueryLanguage, FilterQuery,
+                                   OperationTimeout, ContinueOnError)
+
+        instances = self.ReferenceNames(namespace, InstanceName,
+                                        ResultClass=ResultClass,
+                                        Role=Role)
+
+        return self._open_response(namespace, instances,
+                                   'PullInstancePaths',
+                                   OperationTimeout,
+                                   MaxObjectCount,
+                                   ContinueOnError)
+
+    ####################################################################
+    #
+    #   OpenReferenceInstances server request
+    #
+    ####################################################################
+
+    def OpenReferenceInstances(self, namespace, InstanceName,
+                               ResultClass=None, Role=None,
+                               IncludeClassOrigin=None, PropertyList=None,
+                               FilterQueryLanguage=None, FilterQuery=None,
+                               OperationTimeout=None, ContinueOnError=None,
+                               MaxObjectCount=None):
+        """
+        Implements WBEM server responder for
+        :meth:`~pywbem.WBEMConnection.OpenReferenceInstances`
+        with data from the instance CIM repository.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            Name of the namespace in which the CIM instance will be created
+              (case independent).
```

Comment:
Copy & paste error from CreateInstance. Should use enumeration verbiage.

Reply:
DONE